src/kinematics_plugin.cpp

```diff
@@ -435,20 +362,16 @@ struct BioIKKinematicsPlugin : kinematics::KinematicsBase {
   };*/
 
   virtual bool
-  searchPositionIK(const std::vector<geometry_msgs::Pose> &ik_poses,
+  searchPositionIK(const std::vector<geometry_msgs::msg::Pose> &ik_poses,
                    const std::vector<double> &ik_seed_state, double timeout,
                    const std::vector<double> &consistency_limits,
                    std::vector<double> &solution,
                    const IKCallbackFn &solution_callback,
-                   moveit_msgs::MoveItErrorCodes &error_code,
+                   moveit_msgs::msg::MoveItErrorCodes &error_code,
                    const kinematics::KinematicsQueryOptions &options =
                        kinematics::KinematicsQueryOptions(),
                    const moveit::core::RobotState *context_state = NULL) const {
-    double t0 = ros::WallTime::now().toSec();
-
-    // timeout = 0.1;
-
-    // LOG("a");
+    std::chrono::time_point<std::chrono::system_clock, std::chrono::duration<double>> t0 = std::chrono::system_clock::now();
```

Comment:
Why did you need to specify a unique type here instead just using whatever type `std::chrono::system_clock::now()` already returns?

Reply:
Because `timeout` is specified as a double parameter to this method, which is later added to `t0`